paasta_tools/cli/cmds/remote_run.py

```diff
@@ -0,0 +1,314 @@
+from paasta_tools.cli.utils import run_on_master
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import paasta_print
+from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import PaastaNotConfiguredError
+from paasta_tools.utils import SystemPaastaConfig
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import validate_service_instance
+
+from paasta_tools.native_mesos_scheduler import create_driver_with
+from paasta_tools.native_mesos_scheduler import compose_job_id
+from paasta_tools.native_mesos_scheduler import load_paasta_native_job_config
+
+
+class PaastaAdhocScheduler(Scheduler):
+
+    def __init__ (self, command, service_config, system_paasta_config, dry_run):
+        self.command = command
+        self.service_config = service_config
+        self.system_paasta_config = system_paasta_config
+        self.dry_run = dry_run
+
+
+    def registered(self, driver, framework_id, master_info):
+        paasta_print("Registered with framework id: {}".format(framework_id))
+
+
+    def resourceOffers(self, driver, offers):
+        if self.launched:
+            return
+
+        paasta_print("Recieved resource offers: {}".format([o.id.value for o in offers]))
+        offer = offers[0]
+        task = self.new_task(offer)
+        paasta_print("Launching task {task} "
+                     "using offer {offer}.".format(task=task.task_id.value,
+                                                   offer=offer.id.value))
+
+        self.launched = True
+        if self.dry_run:
+            paasta_print("Not doing anything in dry-run mode."
+                         "Would have launched: {task}".format(task=task))
+            self.status = 0
+            driver.stop()
+        else:
+            # driver.launchTasks(offer.id, tasks)
+            paasta_print("THIS IS NOT A DRILL!"
+                         "Would have launched: {task}".format(task=task))
+            self.status = 0
+            driver.stop()
+
+
+    def statusUpdate(self, driver, update):
+        paasta_print("Mesos Scheduler: task %s is in state %d", update.task_id.value, update.state)
+
+        if update.state == mesos_pb2.TASK_FINISHED:
+            self.status = 0
+        elif update.state == mesos_pb2.TASK_FAILED:
+            # TODO: self.launched = False to retry?
+            self.status = 1
+        elif update.state == mesos_pb2.TASK_LOST or update.state == mesos_pb2.TASK_KILLED:
+            self.launched = False
+
+        if self.status != None:
+            paasta_print("Task %s is finished.", update.task_id.value)
+            for stream in mesos.cli.cluster.files(flist=['stdout','stderr'], fltr=update.task_id.value):
+                print "Printing %s for task %s" % (stream[0].path, update.task_id.value)
+                for line in stream[0].readlines():
+                    print line
+            driver.stop() # is this the right place to stop the driver?
```

Comment:
don't think so, from the docs i got the idea that `driver.run()` (line 256) blocks until driver is stopped. after `.run()` returns we might want to do more stuff depending on exit status, etc.